src/binary/raw_binary_reader.rs

```diff
@@ -478,105 +504,78 @@ impl<R: IonDataSource> RawReader for RawBinaryReader<R> {
         let coefficient = self.read_int(coefficient_size_in_bytes)?;
 
         if coefficient.is_negative_zero() {
-            return Ok(Some(Decimal::negative_zero_with_exponent(exponent)));
-        }
-
-        Ok(Some(Decimal::new(coefficient.value(), exponent)))
-    }
-
-    fn read_big_decimal(&mut self) -> IonResult<Option<BigDecimal>> {
-        read_safety_checks!(self, IonType::Decimal);
-
-        if self.cursor.value.value_length == 0 {
-            return Ok(Some(BigDecimal::new(0i64.into(), 0)));
+            return Ok(Decimal::negative_zero_with_exponent(exponent));
         }
 
-        let exponent_var_int = self.read_var_int()?;
-        let coefficient_size_in_bytes =
-            self.cursor.value.value_length - exponent_var_int.size_in_bytes();
-
-        let exponent = exponent_var_int.value() as i64;
-        let coefficient = self.read_int(coefficient_size_in_bytes)?.value();
-
-        // BigDecimal uses 'scale' rather than 'exponent' in its API, which is a count of the
-        // number of decimal places. It's effectively `exponent * -1`.
-        Ok(Some(BigDecimal::new(coefficient.into(), -exponent)))
+        Ok(Decimal::new(coefficient.value(), exponent))
     }
 
-    fn read_string(&mut self) -> IonResult<Option<String>> {
-        self.string_ref_map(|s: &str| s.into())
+    fn read_string(&mut self) -> IonResult<String> {
+        self.map_string(|s| s.to_owned())
     }
 
-    fn string_ref_map<F, T>(&mut self, f: F) -> IonResult<Option<T>>
+    fn map_string<F, U>(&mut self, f: F) -> IonResult<U>
```

Comment:
PR tour: ultimately the `map_TYPE` methods will be removed in favor of more straightforward ones. They're an artifact of the way the reader's internal buffering works and lifetime/bookkeeping rules that were a bigger deal before Non-Lexical Lifetimes (NLL) became a thing.

See: #335